### DISCUSSION TITLE
Attiny series 2. Can I do precise timing?

### DISCUSSION THREAD

**Discussion by Zibi862** [SpenceKonde/megaTinyCore#846] [Future Development Discussion]
Hi guys, I would like to use Series 2 attinas for precise timing. Is it possible? My knowledge is limited. Attina battery powered.
I wish the processor would do something at the exact time every day. Have any of you done something like that and could a code example help?

**Comment by SpenceKonde** [comment]
With what level of accuracy, and for how long? 

It's as accurate as your timebase is. The internal oscillator is only accurate to, well, see ch 33: Electrical Characteristics section 10: Oscillators and Clocks. it's usually better than those numbers indicate, but there are 86400 seconds per day. An error of 1% is 15 minutes a day. 

To get accurate timing you'd need to use a watch crystal (and the oscillator is rather fiddly - needs a very carefully laid out board. Then you can get accuracy of well, whatever the crystal is capable of, considering crystal aging and imperfections in loading capacitor choice, bad layout, etc. Like, you will not get it to work taking a normal breakout board, nor with anything involving breadboard. 

At one point, before these cores were takign so much time, I was planning a scheme where I would have one device send out a signal at a certain time every day, or several times a day. That device would be internet connected and hence would always be able to find the time accurately, so the other devices could have a bit of inaccuracy and it wouldn't matter

Also worth noting that if you need high speed timing accuracy (hence micros) you instead would need a high accuracy external clock, agauin you are limited to the accuracy oif the timebase. Though with an accurate timebase, you can use input capture to time things to the resolution of a single clock cycle, up to 4.2 billion clock cycles in cascade mode.

**Comment by Zibi862** [comment]
I would like to wake up the processor at 22:00 PM and 6:00 AM.
I've been combining with attiny but it's hard to set u to get precise time.
I think to add PCF8563 tests and I'm glad it has the interrupt pin and I wake up Attiny.
The second option is my old way. The timer I use has 10 different programs it gets with a 2uA display and it works fine too.
![Uploading IMG_20221021_161844.jpg…]()
